### PR TITLE
all: Make systemd restart services on failures

### PR DESCRIPTION
**Pull request by vuntz**
This offers some automatic recovery, which is nice. It turns out it's
also recommended by systemd documentation, in case a service is
long-running, which is the case here.

It's worth mentioning that in the past, the periodic chef-client has
been providing this, to some extent. But this means the recovery could
take up to 15 minutes.

This is of course disabled when a service is managed by Pacemaker, as we
do not want to have systemd interfere with Pacemaker in that case.

### CODE REVIEW

File: chef/cookbooks/barbican/recipes/worker.rb [hunk 1]
```diff
@@ -28,3 +28,6 @@
   action [:enable, :start]
   provider Chef::Provider::CrowbarPacemakerService if use_crowbar_pacemaker_service
 end
+utils_systemd_service_restart "openstack-barbican-worker" do
```
Comment (Itxaka) [inline chef/cookbooks/barbican/recipes/worker.rb]: So hound will complain on the most absurd thing but will ignore the ugliness of not having a new line between blocks. I see how it is, hound ¬_¬

File: chef/cookbooks/postgresql/recipes/server_redhat.rb [hunk 2]
```diff
@@ -90,6 +90,9 @@
   action [:enable, :start]
   provider Chef::Provider::CrowbarPacemakerService if node[:database][:ha][:enabled]
 end
+utils_systemd_service_restart "postgresql" do
+  action node[:database][:ha][:enabled] ? :disable : :enable
```
Comment (Itxaka) [inline chef/cookbooks/postgresql/recipes/server_redhat.rb]: you can reuse the `ha_enabled` attribute here

File: chef/cookbooks/rabbitmq/recipes/default.rb [hunk 3]
```diff
@@ -75,3 +75,6 @@
   action [:enable, :start]
   provider Chef::Provider::CrowbarPacemakerService if node[:rabbitmq][:ha][:enabled]
 end
+utils_systemd_service_restart "rabbitmq-server" do
+  action node[:rabbitmq][:ha][:enabled] ? :disable : :enable
```
Comment (Itxaka) [inline chef/cookbooks/rabbitmq/recipes/default.rb]: same as above

File: chef/cookbooks/postgresql/recipes/server_redhat.rb (hunk 2 above)
Comment (vuntz) [reply chef/cookbooks/postgresql/recipes/server_redhat.rb]: done

File: chef/cookbooks/rabbitmq/recipes/default.rb (hunk 3 above)
Comment (vuntz) [reply chef/cookbooks/rabbitmq/recipes/default.rb]: done

**Pull request merged**